### PR TITLE
make download easier

### PR DESCRIPTION
**Pull request by earringyyr**
Show the download_file on the problem page if needed and show the data_file if the user is SuperUser.
What's more, hide the url in the problem_manege page if it leads to 404 page.
In passing, support that SuperUser can download the data_file by url: /download?type=data&id=problem_id

### CODE REVIEW

File: web/app/controllers/download.php [hunk 1]
```diff
@@ -30,6 +30,14 @@
 			$file_name = "/opt/uoj/judger/uoj_judger/include/testlib.h";
 			$download_name = "testlib.h";
 			break;
+		case 'data':
```
Comment (billchenchina) [inline web/app/controllers/download.php:33]: https://github.com/BIDGroup/UOJ-Edu/blob/master/app/controllers/download.php#L33

**Review by billchenchina — COMMENTED** [review]
Well...
https://github.com/BIDGroup/UOJ-Edu/blob/master/app/controllers/download.php#L33
估计只是碰巧了😂毕竟谁写都会这么写

**Review by billchenchina — APPROVED** [review]
LGTM

**Review by AdemJensen — APPROVED** [review]
LGTM+1

**Pull request closed without merging**